app/src/main/java/com/obsidian/plugins/task_notifier/plugin/ObsidianPluginManager.kt

```diff
@@ -2,56 +2,53 @@ package com.obsidian.plugins.task_notifier.plugin
 
 import android.net.Uri
 import com.google.gson.Gson
-import com.google.gson.internal.LinkedTreeMap
-import com.obsidian.plugins.task_notifier.plugin.dto.v1.ObsidianReminderPluginActualReminderDTO
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonDeserializer
 import com.obsidian.plugins.task_notifier.plugin.dto.v1.ObsidianReminderPluginConfigDTO
 import com.obsidian.plugins.task_notifier.utils.Logger
 import java.text.SimpleDateFormat
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
 import java.util.Date
 
 class ObsidianPluginManager {
-    companion object {
-        const val OBSIDIAN_REMINDER_PLUGIN_CONFIG_FILE_NAME = "data.json"
+  companion object {
+    const val OBSIDIAN_REMINDER_PLUGIN_CONFIG_FILE_NAME = "data.json"
 
-        @JvmStatic
-        fun isInterestedFile(uri: Uri): Boolean {
-            Logger.info(
-                "ObsidianPluginManager uri ${uri.path} contains ${
-                    uri.path?.contains(
-                        OBSIDIAN_REMINDER_PLUGIN_CONFIG_FILE_NAME
-                    )
-                }"
-            )
-            if (uri.path == null || uri.path.isNullOrEmpty()) return false;
-            return uri.path!!.contains(OBSIDIAN_REMINDER_PLUGIN_CONFIG_FILE_NAME)
-        }
+    @JvmStatic
+    fun isInterestedFile(uri: Uri): Boolean {
+      Logger.info(
+        "ObsidianPluginManager uri ${uri.path} contains ${
+          uri.path?.contains(
+            OBSIDIAN_REMINDER_PLUGIN_CONFIG_FILE_NAME
+          )
+        }"
+      )
+      if (uri.path == null || uri.path.isNullOrEmpty()) return false;
+      return uri.path!!.contains(OBSIDIAN_REMINDER_PLUGIN_CONFIG_FILE_NAME)
+    }
 
-        @JvmStatic
-        fun processFile(json: String): List<ObsidianReminderBO> {
-            var result: List<ObsidianReminderBO> = emptyList();
-            val gson = Gson()
-            var obsidianConfig = gson.fromJson(json, ObsidianReminderPluginConfigDTO::class.java)
+    @JvmStatic
+    fun processFile(json: String): ArrayList<ObsidianReminderBO> {
+      val result: ArrayList<ObsidianReminderBO> = ArrayList();
+      val gson = GsonBuilder().registerTypeAdapter(
+        LocalDateTime::class.java,
+        JsonDeserializer<Any?> { json, typeOfT, context ->
+          LocalDateTime.parse(
+            json.asString,
+            DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")
+          )
+        }).create()
+      val obsidianConfig: ObsidianReminderPluginConfigDTO = gson.fromJson(json, ObsidianReminderPluginConfigDTO::class.java)
```

Comment:
move this call to the top of the the function call and use the obsidianConfig.settings.dateTimeFormat for DateTimeFormatter.ofPattern

Reply:
its a little bit complicated to handle it as a date or as a dateTime... :/

Reply:
done